src/Components/Components/src/Microsoft.AspNetCore.Components.csproj

```diff
@@ -22,6 +22,7 @@
   <ItemGroup>
     <Reference Include="Microsoft.Extensions.Logging.Abstractions" />
     <Reference Include="Microsoft.Extensions.DependencyInjection.Abstractions" />
+    <Reference Include="Microsoft.JSInterop" />
```

Comment:
Is this because of the router?

Reply:
yes

Reply:
We want to avoid having a reference to Microsoft.JSInterop here. I would suggest you check how enablenavigationinterception works to see if we can follow a similar approach.

@SteveSandersonMS do you have other thoughts?

Reply:
The way @surayya-MS has addressed this using `IScrollToLocationHash` looks good to me (if we are implementing it on the .NET side, but see comments below).